### PR TITLE
make InputAssignable work with isInputRange

### PR DESCRIPTION
**Pull request by Superstar64**

### CODE REVIEW

File: std/range/interfaces.d [hunk 1]
```diff
@@ -208,6 +208,13 @@ interface RandomAccessInfinite(E) : ForwardRange!E {
 interface InputAssignable(E) : InputRange!E {
     ///
     @property void front(E newVal);
+
+    @property E front();
```
Comment (JackStouffer) [inline std/range/interfaces.d]: `///`

Comment (Superstar64) [inline std/range/interfaces.d]: i'm not sure that's appropriate, the rest of the interfaces only document the functions they add.

Comment (wilzbach) [inline std/range/interfaces.d]: Yes it is,  _all_ public functions should be exposed on the documentation - this includes range functions as well ;-)
(the reason for this is that it's an easy way to show what capabilities are provided by a range.

File: std/range/interfaces.d [hunk 2]
```diff
@@ -208,6 +208,13 @@ interface RandomAccessInfinite(E) : ForwardRange!E {
...
+}
+
```
Comment (wilzbach) [inline std/range/interfaces.d]: `///` here will show the unittest on the dlang.org

File: std/range/interfaces.d (hunk 1 above)
Comment (Superstar64) [inline std/range/interfaces.d]: The other members are already exposed because InputAssignable shows that it inherits from InputRange.

**Comment by Superstar64** [comment]
It appears I didn't the intent of this pr clear enough. isInputRange trys to call InputAssignable!(T).front with no arguments and fails. This prevents passing InputAssignable!T types to any range function.

File: std/range/interfaces.d (hunk 2 above)
Comment (Superstar64) [inline std/range/interfaces.d]: That would be inconsistent. Other similar unittests such as https://github.com/dlang/phobos/blob/master/std/range/interfaces.d#L244 aren't documented.

**Comment by quickfur** [comment]
Wait a second here.  Why does calling `.front` with no arguments fail?!  After all, `InputAssignable!E` inherits from `InputRange!E`, so it *should* have inherited the argumentless `.front` as well.  That it doesn't, contrary to expected class inheritance semantics, seems odd.  Is this a compiler bug?

The proposed fix doesn't seem to be right.  You shouldn't need to declare the argumentless `.front` twice, once in the base interface, once in the derived interface.  If this is a compiler bug, it needs to be fixed in the compiler, not here.

**Comment by quickfur** [comment]
Update: Turns out this is a deliberate design decision, to prevent method hijacking.

The correct fix is to add an `alias`:
````
interface InputAssignable(E) : InputRange!E
{
    @property void front(E e);
    alias front = InputRange!E.front; // overload base interface method
}
````

**Pull request merged**